### PR TITLE
Adds new Entity Type Creation dialog on case page and minor page polish

### PR DESCRIPTION
**Pull request by wssheldon**

### CODE REVIEW

File: src/dispatch/entity_type/flows.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    Args:
+        db_session (Session): The database session to use for entity creation.
+        new_entity_type (EntityType): The newly created EntityType to associate with signals.
+    """
+    # fetch `all` entities that should be associated with all signal definitions
+    entity_types = entity_type_service.get_all(
+        db_session=db_session, scope=EntityScopeEnum.all
+    ).all()
+    entity_types = signal_instance.signal.entity_types + [entity_type]
```
Comment (mvilanova) [inline src/dispatch/entity_type/flows.py:26]: Should we use `append()` instead?

File: src/dispatch/entity_type/views.py [hunk 2]
```diff
@@ -40,13 +45,45 @@ def get_entity_type(db_session: DbSession, entity_type_id: PrimaryKey):
 def create_entity_type(db_session: DbSession, entity_type_in: EntityTypeCreate):
     """Create a new entity."""
     try:
-        entity = create(db_session=db_session, entity_type_in=entity_type_in)
+        entity_type = create(db_session=db_session, entity_type_in=entity_type_in)
     except IntegrityError:
         raise ValidationError(
             [ErrorWrapper(ExistsError(msg="A entity with this name already exists."), loc="name")],
```
Comment (mvilanova) [inline src/dispatch/entity_type/views.py:51]:
```suggestion
            [ErrorWrapper(ExistsError(msg="An entity with this name already exists."), loc="name")],
```

File: src/dispatch/entity_type/views.py [hunk 3]
```diff
@@ -40,13 +45,45 @@ def get_entity_type(db_session: DbSession, entity_type_id: PrimaryKey):
...
         ) from None
-    return entity
+    return entity_type
+
+
+@router.put("/recalculate/{entity_type_id}/{signal_instance_id}", response_model=SignalInstanceRead)
+def recalculate(
+    db_session: DbSession, entity_type_id: PrimaryKey, signal_instance_id: Union[str, PrimaryKey]
+):
+    """Create a new entity."""
```
Comment (mvilanova) [inline src/dispatch/entity_type/views.py:61]: I think this needs to be updated

File: src/dispatch/entity_type/views.py [hunk 4]
```diff
@@ -40,13 +45,45 @@ def get_entity_type(db_session: DbSession, entity_type_id: PrimaryKey):
...
+    entity_type = get(
+        db_session=db_session,
+        entity_type_id=entity_type_id,
+    )
+    if not entity_type:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=[{"msg": "A entity_type with this id does not exist."}],
```
Comment (mvilanova) [inline src/dispatch/entity_type/views.py:69]:
```suggestion
            detail=[{"msg": "An entity type with this id does not exist."}],
```

File: src/dispatch/entity_type/views.py [hunk 5]
```diff
@@ -40,13 +45,45 @@ def get_entity_type(db_session: DbSession, entity_type_id: PrimaryKey):
...
+        )
+
+    signal_instance = get_signal_instance(
+        db_session=db_session,
+        signal_instance_id=signal_instance_id,
+    )
+    if not signal_instance:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=[{"msg": "A signal_instance with this id does not exist."}],
```
Comment (mvilanova) [inline src/dispatch/entity_type/views.py:79]:
```suggestion
            detail=[{"msg": "A signal instance with this id does not exist."}],
```

**Review by mvilanova — APPROVED** [review]

File: src/dispatch/entity_type/flows.py (hunk 1 above)
Comment (wssheldon) [inline src/dispatch/entity_type/flows.py:26]:
```suggestion
    entity_types = signal_instance.signal.entity_types.append(entity_type)
```

File: src/dispatch/entity_type/views.py [hunk 6]
```diff
@@ -40,13 +45,45 @@ def get_entity_type(db_session: DbSession, entity_type_id: PrimaryKey):
...
         ) from None
-    return entity
+    return entity_type
+
+
+@router.put("/recalculate/{entity_type_id}/{signal_instance_id}", response_model=SignalInstanceRead)
+def recalculate(
+    db_session: DbSession, entity_type_id: PrimaryKey, signal_instance_id: Union[str, PrimaryKey]
+):
+    """Create a new entity."""
```
Comment (wssheldon) [inline src/dispatch/entity_type/views.py:61]:
```suggestion
    """Recalculates the associated entities for a signal instance."""
```

**Pull request merged**